great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -525,15 +526,12 @@ def get_domain_records(  # noqa: C901, PLR0912
         if row_condition:
             condition_parser = domain_kwargs.get("condition_parser", None)
 
-            # Ensuring proper condition parser has been provided
-            if condition_parser not in ["python", "pandas"]:
```

Comment:
I removed the `python` condition since it seemed related to `filter_conditions`.